lib/Dialect/CGGI/IR/CGGIOps.td

```diff
@@ -40,12 +40,15 @@ class CGGI_BinaryGateOp<string mnemonic>
   let assemblyFormat = "operands attr-dict `:` qualified(type($output))" ;
 }
 
-def CGGI_AndOp : CGGI_BinaryGateOp<"and"> { let summary = "Logical AND of two ciphertexts."; }
-def CGGI_NandOp : CGGI_BinaryGateOp<"nand"> { let summary = "Logical NAND of two ciphertexts."; }
-def CGGI_NorOp  : CGGI_BinaryGateOp<"nor">  { let summary = "Logical NOR of two ciphertexts."; }
-def CGGI_OrOp  : CGGI_BinaryGateOp<"or">  { let summary = "Logical OR of two ciphertexts."; }
-def CGGI_XorOp : CGGI_BinaryGateOp<"xor"> { let summary = "Logical XOR of two ciphertexts."; }
-def CGGI_XNorOp : CGGI_BinaryGateOp<"xnor"> { let summary = "Logical XNOR of two ciphertexts."; }
+def CGGI_AndOp : CGGI_BinaryOp<"and"> { let summary = "Logical AND of two ciphertexts."; }
+def CGGI_NandOp : CGGI_BinaryOp<"nand"> { let summary = "Logical NAND of two ciphertexts."; }
+def CGGI_NorOp  : CGGI_BinaryOp<"nor">  { let summary = "Logical NOR of two ciphertexts."; }
+def CGGI_OrOp  : CGGI_BinaryOp<"or">  { let summary = "Logical OR of two ciphertexts."; }
+def CGGI_XorOp : CGGI_BinaryOp<"xor"> { let summary = "Logical XOR of two ciphertexts."; }
+def CGGI_XNorOp : CGGI_BinaryOp<"xnor"> { let summary = "Logical XNOR of two ciphertexts."; }
+def CGGI_AddOp : CGGI_BinaryOp<"add"> { let summary = "Arithmetic addition of two ciphertexts."; }
+def CGGI_MulOp : CGGI_BinaryOp<"mul"> { let summary = "Arithmetic multiplication of two ciphertexts."; }
```

Comment:
```suggestion
def CGGI_MulOp : CGGI_BinaryOp<"mul"> { 
  let summary = "Arithmetic multiplication of two ciphertexts."; 
  let description = [{
    While CGGI does not have a native multiplication operation,
    some backend targets provide a multiplication
    operation that is implemented via a sequence
    of other atomic CGGI ops. When lowering to
    backends that do not have this, one must lower
    to this op the appropriate CGGI ops.
  }];
}
```

Perhaps it would be worth filing an issue for implementing a lowering of this op to other CGGI atomic ops? That might be a "good first issue" for some newcomers.